src/main/java/com/internet/shop/controller/LoginController.java

```diff
@@ -0,0 +1,37 @@
+package com.internet.shop.controller;
+
+import com.internet.shop.exception.AuthException;
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.User;
+import com.internet.shop.security.AuthenticationService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+@WebServlet("/login")
+public class LoginController extends HttpServlet {
+    public static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final AuthenticationService authenticationService =
+            (AuthenticationService) injector.getInstance(AuthenticationService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("username");
+        try {
+            String password = req.getParameter("password");
+            User user = authenticationService.login(login, password);
+            HttpSession httpSession = req.getSession();
+            httpSession.setAttribute("userId", USER_ID);
```

Comment:
so are we using session here?

Reply:
Here no. But if i will not pass that attribute to session my logic for homepage will be broken and we take that attribute in filter